src/codegen/c.zig

```diff
@@ -200,6 +229,27 @@ pub const DeclGen = struct {
                 }
             },
             .Bool => return writer.print("{}", .{val.toBool()}),
+            .Optional => {
+                var opt_buf: Type.Payload.ElemType = undefined;
+                const child_type = t.optionalChild(&opt_buf);
+                if (t.isPtrLikeOptional()) {
+                    return dg.renderValue(writer, child_type, val);
+                }
+                try writer.writeByte('(');
+                try dg.renderType(writer, t);
+                if (val.tag() == .null_value) {
+                    try writer.writeAll("){ .is_null = true }");
+                } else {
+                    try writer.writeAll("){ .is_null = false, .payload = ");
+                    try dg.renderValue(writer, child_type, val);
+                    try writer.writeAll(" }");
+                }
+            },
+            .ErrorSet => {
+                const payload = val.castTag(.@"error").?;
+                // error values will be #defined at the top of the file
+                return writer.print("zig_error_{s}", .{payload.data.name});
+            },
             else => |e| return dg.fail(dg.decl.src(), "TODO: C backend: implement value {s}", .{
```

Comment:
I think you are missing error union here. I will do it as I am also making tests for this right now.

Reply:
Actually this requires some more infrastructure around runtime error comparison/wrapping/unwrapping error unions. I will try to do this.

Reply:
Could you actually rebase this? It needs to be done before I can contribute to it. I do not think I can rebase it myself.

Reply:
> Could you actually rebase this?

Sure

>I think you are missing error union here. 

We didn't have error union values anywhere in the compiler when I wrote this and I didn't want to implement them in this pr.

Reply:
https://github.com/Vexu/zig/pull/2